jerry-core/ecma/base/ecma-globals.h

```diff
@@ -985,10 +1007,10 @@ typedef struct
 
     lit_utf8_size_t long_utf8_string_size; /**< size of this long utf-8 string in bytes */
     uint32_t uint32_number; /**< uint32-represented number placed locally in the descriptor */
-    lit_magic_string_id_t magic_string_id; /**< identifier of a magic string */
-    lit_magic_string_ex_id_t magic_string_ex_id; /**< identifier of an external magic string */
+    uint32_t magic_string_id; /**< identifier of a magic string */
+    uint32_t magic_string_ex_id; /**< identifier of an external magic string */
```

Comment:
The problem was that their storage size was unknown before. Now uint32_number, magic_string_id, and magic_string_ex_id uses the same storage type (common_uint32_field) and this opens up several optimizations.